### DISCUSSION TITLE
Does user PLATOMF REALLY require SAL=377B validation???

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#111] [General]
SAV=0
SAC=0

App perms seem a bit ridiculous too.  All on but confer and icf.  User plato seems to have the same app permissions as well, but with SAL=1, user sys as well.

User systemx has all sal and sav, but sac=0.

I am getting up on this, but some of the validation file things...  even if not a major issue now, minor issues here can cause big ones, yeah, passwords are six chars and can be brute-forced in a millisecond, but....

**Comment by kej715** [comment]
On systems installed from ready-to-run packages, PLATOMF has SAL=1.  Where
are you seeing SAL=377B? Or, are you asking whether PLATOMF's SAL should be
changed to 377B?

**Reply by KC5VDJ** [reply]
Apparently that is what the source install did (basic, then built up from there to full/cybis/tms in several saved versions) The bottom app on each side disabled, with all others on, might have been a carried over error in cut and paste?  Just guessing.  It would seem that they shouldn't have such perms, but having only just noticed it, I have yet to go through any docs / files that might explain it.  Am I missing something here?

**Comment by kej715** [comment]
The user PLATOMF and associated privileges are created in a source install
by NOS2.8.7/opt/cybis-users.job. It is not modified anywhere else. In this
job, SAL is first set explicitly to NUL to clear all bits, and then it is
set to LVL0 (as in "SAL = LVL0"). LVL0 is 1. It seems unlikely that the
source install would cause SAL to be set to 377B as there's only one place
where PLATOMF is created or altered, and it explicitly clears the field
before then setting it to 1.

Note that CYBISMF, which shares user index 377773B with PLATOMF, -is-
created with SAL=377B. These privileges were established in cybis-users.job
by William Schaub who, presumably, was following the original CYBIS
installation documentation. It's not clear why PLATOMF and CYBISMF would
have difference SAL values, but this is probably deliberate and in
accordance with original CDC recommendations.

**Comment by wschaub** [comment]
I copied Tom Hunter's CYBIS Release 1 kit by dumping the validation file to source when I made the job to setup the CYBIS related users.  I don't know if SAL having all the levels set is needed or not. Isn't that only applicable to multi-level security anyway? (we don't enable that and I doubt anyone outside the government ever used the multi-level security support)

**Comment by kej715** [comment]
AFAIC, no customer sites other than, possibly, military sites or
spy agencies used multi-level security. And, of course, you're correct that
we don't enable it in the install-from-scratch or ready-to-run systems.

**Reply by KC5VDJ** [reply]
Ok.  I've only experienced a site that did, and with enterprise other experience, I always look at potential permissions issues.  Without external encryption anyway, and rewritten validation (necessary os / fs structure changes, aka incompat unless shoehorned into RFUs)... I consider security to be largely host-based with this one, but, it always pays to know issues with the hosted.

PS: want a listing showing spectre vulnerable code sections?  Many..  trying to clean the lint (Wall) too; going to -ipo (as opposed to -ip) would be a worthy goal (causes probs currently).  Intel compiler level 5 reporting.

**Reply by KC5VDJ** [reply]
Actually, this is using Intel classic (need to migrate to the oneAPI version still), this will produce the reports:

[dtcyber.intel.options_094227.txt](https://github.com/kej715/DtCyber/files/13833282/dtcyber.intel.options_094227.txt)

/permissive /MP /FR"x64\Release\Intermediate\" /GS /GA /Wall /QxCORE-AVX2 /Gy /Zc:wchar_t /Qconditional-branch:pattern-report /Zi /O3 /Qopt-report:5 /Ob2 /Fd"x64\Release\Intermediate\vc143.pdb" /Zc:inline /fp:fast=2 /Quse-intel-optimized-headers /tune:haswell /D "WIN32" /D "_CONSOLE" /D "_CRT_SECURE_NO_WARNINGS" /D "_CRT_NONSTDC_NO_DEPRECATE" /D "IdleThrottle" /D "NDEBUG" /D "_VC80_UPGRADE=0x0600" /D "_MBCS" /Qip /GF /GT /Qoffload-arch:haswell:visa3.1 /Zc:forScope /Qopt-matmul /arch:CORE-AVX2 /Oi /MT /FC /Fa"x64\Release\Intermediate\" /EHsc /nologo /Qparallel /Fo"x64\Release\Intermediate\" /Qprof-dir "x64\Release\Intermediate\" /FAcs /Ot /Fp"x64\Release\Intermediate\DtCyber.pch"

From what I am seeing, some of the spectre fixes could require restructuring (major release).  Wirth was right after all.
